extensions/shared/external-media/sources/with-media.js

```diff
@@ -186,14 +186,27 @@ export default function withMedia() {
 					path: apiUrl,
 					method: 'POST',
 					data: {
+						external_ids: items.map( item => item.guid ), // WPCOM.
```

Comment:
Could we possibly change the WPCOM api so that it reads from `media` instead of `external_ids`?

Reply:
It would require versioning the API, which this PR is aiming to avoid. I think had we been involved in creating this feature from the start we'd probably shaped the new API endpoint to closer resemble the existing one, but that 🚢  has sailed. 

Providing compatibility doesn't come at a huge cost though, so I'm not too worried about it.